### PR TITLE
chore(v3): Update `release.config.js`

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Updates the `release.config.js` file for v3:
- Changes `release/2.*` branch pattern to publish on the `zowe-v2-lts` channel
- Adds `release/3.*` branch pattern to publish v3 releases to the `latest` channel

## Release Notes

Milestone: vNext

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [x] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki

### CODE REVIEW

File: .github/release.config.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
 module.exports = {
     branches: [
         {
-            name: "main",
-            channel: "latest",
-            level: "minor",
+            name: "v1-lts",
+            channel: "zowe-v1-lts",
+            level: "patch",
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply .github/release.config.js:6]: yeah, feel free to remove it from here.
Hopefully we don't need to add it back in the future 😋

**Review by zFernand0 — APPROVED** [review]

**Review by JillieBeanSim — COMMENTED** [review]
I'm curious about these changes, wouldn't 2.x be the v2-lts branch vs `release/2.*` (not super important since just a name but v2-lts branch was already created with protection rules) and we have the `main` plus `release/3.*` for v3 work, is the release/3.* just taking place of maintenance?

**Review by zFernand0 — APPROVED** [review]

**Comment by traeok** [comment]
> I'm curious about these changes, wouldn't 2.x be the v2-lts branch vs `release/2.*` (not super important since just a name but v2-lts branch was already created with protection rules) and we have the `main` plus `release/3.*` for v3 work, is the release/3.* just taking place of maintenance?

For `release/2.*`, this allows us to push patch releases for a v2 version in case we need to release a fix. For example, the `release/2.18` branch represents our last v2 release, so if we merge changes into that branch, it will allow us to publish new v2 patches from that branch.

For `release/3.*`, this allows the same but for v3 releases. It will allow us to make fixes to a v3 release without releasing it with new features already merged into `main` - those release branches would act similarly to maintenance as you mentioned.

File: .github/release.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (traeok) [reply .github/release.config.js]: Sorry about that! Addressed in latest commit

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
if the branch names are to be changed for the release workflows then the branches need to be created with branch protection rules before we merge these changes in.  @traeok do you have permissions to set branch protection rules? If not @zFernand0 or I can help get this set up.

**Comment by traeok** [comment]
@JillieBeanSim Thanks for the reminder, I've added the new protection rules - I matched the settings for the other protected branches.

**Review by JillieBeanSim — APPROVED** [review]
Thanks @traeok for addressing comments

**Pull request merged**